File: nmaas-portal/src/app/model/appinstancestatus.ts
export enum AppInstanceState {
  REQUESTED,
  VALIDATION,
  PREPARATION,
  CONNECTING,
  CONFIGURATION_AWAITING,
  DEPLOYING,
  RUNNING,
  UNDEPLOYING,
  DONE,
  FAILURE,
  UNKNOWN
}

export class AppInstanceStatus {
  public appInstanceId: number = undefined;
  public state: AppInstanceState = undefined;
  public previousState: AppInstanceState = undefined;
  public details: string = undefined;
  public userFriendlyDetails: string = undefined;
  public userFriendlyState: string = undefined;
}

File: src/main/java/net/geant/nmaas/portal/domain/AppInstanceStatus.java
package net.geant.nmaas.portal.domain;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Setter
@Getter
@Builder
public class AppInstanceStatus {
	private Long appInstanceId;
	private AppInstanceState state;
	private AppInstanceState previousState;
	private String details;
	private String userFriendlyDetails;
	private String userFriendlyState;
}
